src/OrleansProviders/Streams/Common/PooledCache/FixedSizeObjectPool.cs

```diff
@@ -75,8 +84,25 @@ public override void Free(T resource)
         {
             lock (locker)
             {
+                if(this.usedObjects.Contains(resource))
+                    this.usedObjects.Remove(resource);
```

Comment:
Contains enumerates, and Remove enumerates once again. Just Remove directly without checking if it's there